packages/edit-post/src/components/header/contextual-toolbar/index.js

```diff
@@ -0,0 +1,74 @@
+/**
+ * External dependencies
+ */
+import classnames from 'classnames';
+
+/**
+ * WordPress dependencies
+ */
+import {
+	BlockToolbar,
+	privateApis as blockEditorPrivateApis,
+} from '@wordpress/block-editor';
+import { useEffect, useRef } from '@wordpress/element';
+import { Button, Popover } from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+import { next, previous } from '@wordpress/icons';
+
+/**
+ * Internal dependencies
+ */
+import { unlock } from '../../../lock-unlock';
+
+const { useHasBlockToolbar } = unlock( blockEditorPrivateApis );
+
+function ContextualToolbar( {
+	blockSelectionStart,
+	isCollapsed,
+	toggleCollapse,
+} ) {
+	const blockToolbarRef = useRef();
+	const hasBlockToolbar = useHasBlockToolbar();
+
+	const hasBlockSelection = !! blockSelectionStart;
+
+	useEffect( () => {
+		// If we have a new block selection, show the block tools
+		if ( blockSelectionStart ) {
+			toggleCollapse( false );
+		}
+	}, [ blockSelectionStart, toggleCollapse ] );
+
+	if ( ! hasBlockToolbar ) {
+		return null;
+	}
+
+	return (
+		<>
+			<div
+				className={ classnames( 'selected-block-tools-wrapper', {
```

Comment:
I know this predates this PR but this class (and the next one on this file) don't follow our CSS classnames naming guidelines.